substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -109,6 +109,15 @@ fn crate_metadata(cargo_manifest: &Path) -> Metadata {
 	crate_metadata
 }
 
+/// Keep the build directories separate so that when switching between the
+/// targets we won't trigger unnecessary rebuilds.
+fn build_subdirectory(target: RuntimeTarget) -> &'static str {
```

Comment:
Why not keep it as inherent function?

Reply:
It's local to `wasm_project.rs`. Let's reconsider if this ever changes.